addons/mrp_account/tests/test_analytic_account.py

```diff
@@ -224,3 +224,30 @@ def test_changing_mo_analytic_account(self):
         # Check that the AA lines are recreated correctly if we delete the AA, save the MO, and assign a new one
         mo.analytic_account_id = self.analytic_account
         self.assertEqual(len(mo.move_raw_ids.analytic_account_line_id), 1)
+
+    def test_add_wo_analytic_no_company(self):
+        """Test the addition of work orders to a MO linked to
+        an analytic account that has no company associated
+        """
+        # Create an analytic account and remove the company
+        analytic_account_no_company = self.env['account.analytic.account'].create({
+            'name': 'test_analytic_account_no_company'
+        })
+        analytic_account_no_company.company_id = None
+
+        # Create a mo linked to an analytic account with no associated company
+        mo_no_company = self.env['mrp.production'].create({
+            'product_id': self.product.id,
+            'analytic_account_id': analytic_account_no_company.id,
+            'product_uom_id': self.bom.product_uom_id.id,
+        })
+
+        # Add a work order, this creates a traceback without the fix
+        self.env['mrp.workorder'].create({
+            'name': 'Work_order',
+            'workcenter_id': self.workcenter.id,
+            'product_uom_id': self.bom.product_uom_id.id,
+            'production_id': mo_no_company.id,
+        })
+        # Verify the work order has been added
```

Comment:
no need to add comments in this case since it's obvious from the code :wink: 
```suggestion
        self.env['mrp.workorder'].create({
            'name': 'Work_order',
            'workcenter_id': self.workcenter.id,
            'product_uom_id': self.bom.product_uom_id.id,
            'production_id': mo_no_company.id,
        })
```